src/libponyc/pass/syntax.c

```diff
@@ -1259,6 +1260,29 @@ static ast_result_t syntax_annotation(pass_opt_t* opt, ast_t* ast)
 }
 
 
+static ast_result_t syntax_as(pass_opt_t* opt, ast_t* ast)
+{
+  pony_assert(ast_id(ast) == TK_AS);
+  AST_GET_CHILDREN(ast, expr);
+
+  switch (ast_id(expr))
+  {
+    case TK_INT:
+    case TK_FLOAT:
+      ast_error(opt->check.errors, expr,
+        "Cannot cast uninferred numeric literal");
+      ast_error_continue(opt->check.errors, expr,
+        "To give a numeric literal a specific type, "
+        "use constructor of primitive");
```

Comment:
Good error message!

But I think we can be a little more clear here to give the user a little extra help - I worry that `primitive` may be confusing because the user may not be thinking of the numeric types as primitives (they are primitives, but that's a bit of a detail we might not expect them to know)

So, instead of "use constructor of primitive", I'd say "use the constructor of that numeric type".